### PR TITLE
IGNITE-17887 CMG RAFT commands converted to Transferable classes

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-17887

### CODE REVIEW

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterState.java [hunk 1]
```diff
@@ -32,76 +33,32 @@
...
-            return false;
-        }
-        ClusterState state = (ClusterState) o;
-        return cmgNodes.equals(state.cmgNodes) && msNodes.equals(state.msNodes) && igniteVersion.equals(state.igniteVersion)
-                && clusterTag.equals(state.clusterTag);
-    }
+    /**
+     * Returns a version of Ignite nodes that comprise this cluster.
+     */
+    String version();
```
Comment (tkalkirill) [inline modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterState.java:51]: Why string and not byte array?

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/network/messages/CmgMessageGroup.java [hunk 2]
```diff
@@ -53,4 +53,27 @@ public class CmgMessageGroup {
      * Message type for {@link SuccessResponseMessage}.
      */
     public static final short SUCCESS_RESPONSE = 6;
+
+    /**
+     * Message types for RAFT commands.
+     */
+    public interface Commands  {
```
Comment (tkalkirill) [inline modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/network/messages/CmgMessageGroup.java:60]: Can you add a description to the commands?

File: modules/cluster-management/src/test/java/org/apache/ignite/internal/cluster/management/raft/AbstractClusterStateStorageManagerTest.java [hunk 3]
```diff
@@ -213,4 +216,22 @@ void testValidatedNodes() {
 
         assertThat(storageManager.getValidatedNodeIds(), containsInAnyOrder("node2"));
     }
+
+    private static ClusterTag clusterTag(String cluster) {
```
Comment (tkalkirill) [inline modules/cluster-management/src/test/java/org/apache/ignite/internal/cluster/management/raft/AbstractClusterStateStorageManagerTest.java:220]: Can these methods be taken out in a separate auxiliary class?

File: modules/cluster-management/src/test/java/org/apache/ignite/internal/cluster/management/raft/CmgRaftGroupListenerTest.java [hunk 4]
```diff
@@ -63,22 +62,24 @@ void tearDown() throws Exception {
      */
     @Test
     void testValidatedNodeIds() {
-        var state = new ClusterState(
-                List.of("foo"),
-                List.of("bar"),
-                IgniteProductVersion.CURRENT_VERSION,
-                new ClusterTag("cluster")
-        );
+        ClusterTag clusterTag = MSG_FACTORY.clusterTag().clusterName("cluster").clusterId(UUID.randomUUID()).build();
```
Comment (tkalkirill) [inline modules/cluster-management/src/test/java/org/apache/ignite/internal/cluster/management/raft/CmgRaftGroupListenerTest.java:65]: Same

File: modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java [hunk 5]
```diff
@@ -1359,6 +1367,17 @@ public <T> T[] readObjectArray(MessageCollectionItemType itemType, Class<T> item
...
+    }
+
+    @Override
+    public <C extends Set<?>> C readSet(MessageCollectionItemType itemType, MessageReader reader) {
+        return readCollection0(itemType, reader, HashSet::new);
+    }
+
+    @Nullable
+    private <C extends Collection<?>> C readCollection0(MessageCollectionItemType itemType, MessageReader reader,
+            IntFunction<Collection<Object>> ctor) {
```
Comment (tkalkirill) [inline modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java:1380]: Lacks description.

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/network/messages/CmgMessageGroup.java (hunk 2 above)
Comment (ibessonov) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/network/messages/CmgMessageGroup.java:60]: Yes

File: modules/cluster-management/src/test/java/org/apache/ignite/internal/cluster/management/raft/AbstractClusterStateStorageManagerTest.java (hunk 3 above)
Comment (ibessonov) [reply modules/cluster-management/src/test/java/org/apache/ignite/internal/cluster/management/raft/AbstractClusterStateStorageManagerTest.java:220]: Sure

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterState.java (hunk 1 above)
Comment (ibessonov) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterState.java:51]: String is more convenient and only, like, 2-4 bytes more than byte[]. Not worth it

**Review by tkalkirill — APPROVED** [review]

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/network/messages/CmgMessageGroup.java [hunk 6]
```diff
@@ -53,4 +62,54 @@ public class CmgMessageGroup {
...
+        /**
+         * Messae type for {@link InitCmgStateCommand}.
```
Comment (sergey-chugunov-1985) [inline modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/network/messages/CmgMessageGroup.java:71]: Typo in word `Message`, please fix it.

**Review by sergey-chugunov-1985 — CHANGES_REQUESTED** [review]

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/network/messages/CmgMessageGroup.java (hunk 6 above)
Comment (ibessonov) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/network/messages/CmgMessageGroup.java]: Thank you! Done

**Pull request merged**